Fix Visual mode flip on project search

MostlyKIGuess opened the pull request:
- Prevent incorrect visual-mode activation when opening project search results by ensuring selections made before the editor gains focus

### Some Diagnosis:
- The results editor got created before collapse_matches was initialized -> giving  range_for_match a full selection instead of a collapsed cursor. 
- When the editor finally focused and Vim synced settings, it saw that non-empty selection -> flip into Visual mode.
- From what I understand, it happened only on the first run because: the editor is _not_ focused yet when created, so `sync_vim_settings` is _not_ called. This means `collapse_matches` is still `false` (the default). ( so when called from [project_search.rs](crates/search/src/project_search.rs) L1515 collapse matches is false and later on it focuses on the editor and due to that it changes to visual. )

Closes #43878 

Release Notes:

- Fixed vim mode incorrectly switching to Visual mode on first project search

MostlyKIGuess committed:
fix: set collapse_matches before registering Vim addon

- Prevent incorrect visual-mode activation when opening project search
  results by ensuring selections made before the editor gains focus

ovidius72 commented:
> Intended Behavior [demo](https://github.com/user-attachments/assets/7677bdaa-e5df-4b47-82a0-cda73e426c6f)

Looks great. Did you do some tests when you go back to the file where the search started from? I found it also to  be in visual mode.

dinocosta committed:
refactor(vim): sync editor settings immediately when activating vim

Previously, vim settings like `collapse_matches` were only synced to an
editor when it received focus via `EditorEvent::FocusedIn`. This could
leading to timing issues with, for example, project search, as when the
results editor was created, `collapse_matches` remained false, so
selecting the first match would create a non-empty selection, causing
vim to switch to Visual mode.

This commit refactors `sync_vim_settings` to extract settings into a
`VimEditorSettingsState` struct and apply them via
`sync_vim_settings_to_editor`. This allows `Vim::activate` to sync
settings immediately without re-entrantly borrowing the editor.

Also adds regression tests for both vim and helix modes to ensure
project search opens in the correct mode (Normal/HelixNormal).

dinocosta reviewed (approved):
@MostlyKIGuess Thank you so much for tackling this! Appreciate the detailed diagnosis in the PR's description, helps frame the suggested changes ✌️

I noticed that, since we're updating `Vim::activate` to set the `collapse_matches` field, we might as well try to actually sync all settings, like `Vim.sync_settings` is doing, so I'm wrapping up a commit with these changes.

Will merge soon! 🙂

dinocosta committed:
Merge branch 'main' into deploy-search-fix

dinocosta committed:
fix(vim): fix failing windows test

dinocosta committed:
fix(vim): fix failing windows test
